bioptim/dynamics/integrator.py

```diff
@@ -763,28 +768,28 @@ def dxdt(
         """
 
         # Total number of variables for one finite element
-        states_end = self._d[0] * states[0]
+        states_end = self._d[0] * states[1]
         defects = []
         for j in range(1, self.degree + 1):
             # Expression for the state derivative at the collocation point
-            xp_j = 0
-            for r in range(self.degree + 1):
-                xp_j += self._c[r, j] * states[r]
+            xp_j = self._c[0, j] * states[0]
+            for r in range(1, self.degree + 1):
+                xp_j += self._c[r, j] * states[r + 1]
```

Comment:
Please investigate these

```python
xp_j = 0
for r in range(1, self.degree + 1):
	xp_j += self._c[r + 1 if r == 1 else r, j] * states[r]
```			
				
```python
xp_j = 0
for r in range(self.degree + 1):
	if r == 1:
		continue
	xp_j += self._c[r + 1 if r < 1 else r, j] * states[r]
```


```python
xp_j = self._c[0, j] * states[0]
for r in range(2, self.degree + 2):
	# We need to subtract 1 to r because we skipped it starting at 2
	xp_j += self._c[r - 1, j] * states[r]
```

Reply:
Done.